specification/proto-design-goals.md

```diff
@@ -0,0 +1,19 @@
+# Design Goals for OpenTelemetry Wire Protocol
+
+We want to design a telemetry data exchange protocol that has the following characteristics:
+
+- Be suitable for use between all of the following node types: instrumented applications, telemetry backends, local agents, stand-alone collectors/forwarders.
+
+- Have high reliability of data delivery and clear visibility when the data cannot be delivered.
+
+- Have low CPU usage for serialization and deserialization.
```

Comment:
nit: I prefer `resource` instead of `CPU` since we are at high-level.

Reply:
I think it is worth listing CPU and Memory separately. They are related but somewhat distinct with possibly different solutions. Especially the memory, I wanted to mention it in a separate paragraph so that we realize the specific pattern of allocating and using memory can be quite unfriendly to GCs.